geoh5py/shared/utils.py

```diff
@@ -262,14 +262,18 @@ def are_objects_similar(obj1, obj2, ignore):
     return attributes1 == attributes2
 
 
-def compare_entities(
+def compare_entities(  # pylint: disable=too-many-branches
```

Comment:
disabling pylint here and there defeats the purpose of using to help maintaining a clean code base.
Current setting is already at 13 branches max...
That is a hint that this function needs to be split

Reply:
Ok, I split the function up into a bunch of compare_x methods